pkg/k8sutil/create.go

```diff
@@ -161,3 +168,60 @@ func parseJSONManifest(data []byte) ([]manifest, error) {
 	}
 	return manifests, nil
 }
+
+// CreateOrUpdateNamespace creates the release namespace or updates the namespace
+// if it already exists.
+func CreateOrUpdateNamespace(ns Namespace, nsclient corev1typed.NamespaceInterface) error {
+	if ns.Name == "" {
+		return fmt.Errorf("namespace name can't be empty")
+	}
+
+	// Ensure the namespace in which we create release and resources exists.
+	_, err := nsclient.Create(context.TODO(), &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        ns.Name,
+			Labels:      ns.Labels,
+			Annotations: ns.Annotations,
+		},
+	}, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+
+	if apierrors.IsAlreadyExists(err) {
+		if err := updateNamespace(ns, nsclient); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// updateNamespace updates the namespace.
+func updateNamespace(namespace Namespace, nsclient corev1typed.NamespaceInterface) error {
+	if namespace.Name == "" {
```

Comment:
I think we can skip this check, as this is unexported function. And we should have unit test for exported one checking that.